### PR TITLE
A task to run S3 Java tests against RGW

### PR DESCRIPTION
**Pull request by adamyanova**
A task to run S3 Java tests against RGW

- option to download test suite from a custom repository

- imports SSL certificates to java keystore

- reads data from config file and create RGW users

- installs required packages and runs the tests

### CODE REVIEW

File: qa/tasks/s3tests_java.py [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+        - s3tests-java: [client.0]
+
+    To pass extra arguments to gradle (e.g. to run a certain test, 
+    specify a different repository and branch for the test suite, 
+    run in debug mode or forward the gradle output to a log file)::
+
+        tasks:
+        - ceph:
+        - rgw: [client.0]
+        - s3tests_java_local:
```
Comment (alimaredia) [inline qa/tasks/s3tests_java.py]: Should `s3tests_java_local` here be `s3tests-java`?

**Comment by alimaredia** [comment]
This looks good to me. Could you run it through teuthology suite testing this task against your revised branch of java s3tests that you opened up a pull request for?

**Comment by adamyanova** [comment]
Here is a log with successful runs in teuthology-suite:
[http://pulpito.ceph.com/adamyanova-2018-08-03_12:44:21-rgw:verify-master-distro-basic-smithi/](http://pulpito.ceph.com/adamyanova-2018-08-03_12:44:21-rgw:verify-master-distro-basic-smithi/)

The task has been initiated from the wip-s3java-suite branch due to the extra arguments required before the merging of [https://github.com/ceph/java_s3tests/pull/1](https://github.com/ceph/java_s3tests/pull/1) into the official ceph repository. 

The two tasks are identical and could be checked with
    git diff wip-s3java wip-s3java-suite qa/tasks/s3tests_java.py 

The extra lines in 
   git diff wip-s3java wip-s3java-suite qa/suites/rgw/verify/tasks/s3tests-java.yaml 

allow the use of a custom repository for the tests suite. The default is [https://github.com/ceph/java_s3tests.git](https://github.com/ceph/java_s3tests.git)

**Comment by adamyanova** [comment]
jenkins retest this please

**Comment by mattbenjamin** [comment]
unstale me, my man

**Comment by mattbenjamin** [comment]
@alimaredia didn't the SSL issue get resolved?

**Comment by adamyanova** [comment]
jenkins retest this please

**Comment by adamyanova** [comment]
jenkins retest this please

**Comment by adamyanova** [comment]
jenkins retest this please

**Comment by adamyanova** [comment]
jenkins retest this please

**Comment by alimaredia** [comment]
this PR is not stale

File: .gitignore [hunk 2]
```diff
@@ -68,3 +68,9 @@ GTAGS
 # dashboard
 /src/pybind/mgr/dashboard/frontend/src/environments/environment.ts
 /src/pybind/mgr/dashboard/frontend/src/environments/environment.prod.ts
+.classpath
```
Comment (alimaredia) [inline .gitignore]: @adamyanova why are these being added to .gitignore?

Comment (adamyanova) [reply .gitignore]: @alimaredia  Which one do you refer to?
The *ts files come from the master branch. I myself have not explicitly added them. 
The .classpath is created from the editor I used, hence I ignored it.

Comment (alimaredia) [reply .gitignore]: I was talking about the `.classpath`, `.project`, and the ones below that. Can the changes to this file be removed?

Comment (adamyanova) [reply .gitignore]: done

**Comment by alimaredia** [comment]
unstale this.

**Comment by mattbenjamin** [comment]
@alimaredia what is blocking this?

**Pull request merged**